### PR TITLE
fix(devtools): harden lint-capability stripComments with a char-scan tokenizer (closes #414)

### LINKED ISSUE #414: Harden lint-capability stripComments against // inside string/regex literals
Follow-up from PR #413 review (gemini-code-assist, on the #412 code).

`stripComments` in \`packages/devtools/src/methodology/capability-lint.ts\` removes \`// …\` line comments via a regex with a single guard (`//` not preceded by `:`, to protect URLs). It can still over-strip a `//` that appears inside a **string literal** (e.g. \`const msg = "Choose // to proceed"\`) or a **regex literal** (e.g. \`/^\/\//\`).

**Impact: low.** The checker only extracts module specifiers from `import/export … from "…"`, `require("…")`, and `import("…")`. Over-stripping can at most cause a *false negative* (an import on the same physical line *after* such a literal could be missed) — never a false positive or a crash. No real addon hits this today (the full 29-addon sweep is clean).

**Possible hardening:** match strings/regexes and comments together and only strip the comment alternatives (keeping string/regex contents), or do a minimal char-scan tokenizer. Weigh against KISS — a fragile regex rewrite may introduce more risk than it removes.

### PR DESCRIPTION
**Pull request by laofahai**
## What

Closes #414 (P3 hardening follow-up from #413 review). Replaces the two-regex `stripComments` in `packages/devtools/src/methodology/capability-lint.ts` with a **single-pass, dependency-free character-scan tokenizer**.

The old version guarded only `//` preceded by `:` (to protect `https://`), so it still over-stripped a `//` that lived inside a **string**, **template**, or **regex** literal.

## How

The tokenizer walks the source one char at a time and:
- removes `// …` line comments (to end of line) and `/* … */` block comments (multi-line; replaced with a single space so a `import /* x */ { a } from "…"` still parses);
- preserves the **full contents** of `'…'`, `"…"`, `` `…` ``, and `/…/` literals — including any inner `//`, `/*`, quotes, backslashes — honouring escapes (`\"`, `\\`, `\/`, …) and regex `[…]` char classes;
- uses a pragmatic **regex-vs-division heuristic**: a `/` starts a regex only when the previous *significant* (non-whitespace, **non-comment**) char is a punctuator, the start of input, or a regex-permitting keyword (`return`, `typeof`, `case`, …). Documented limits: it does not track keywords reused as identifiers, and treats template literals as opaque (no `${…}` recursion) — both safe here.

`stripComments` and `extractImportSpecifiers` are now `export`ed from the source file (NOT from the package public index) purely so the tokenizer can be unit-tested directly.

## Why it matters

- Eliminates the #414 **false negative** (a real import dropped when placed after a string-with-`//` on the same physical line).
- Also closes a related **false positive**: a core-internal path named in prose in a comment trailing a regex-returning line (`return /re/ // see @linchkit/core/src/foo`) used to leak in as a phantom import.

## Verification

- `bun test packages/devtools/__tests__/capability-lint.test.ts` → **37 pass / 0 fail** (21 stripComments + extractImportSpecifiers cases, incl. the keyword-after-`return` regex path and block-comment heuristic transparency the old regex could not express).
- Full `bun test` → 6346 pass, **0 named failures** (the reported "1 fail/1 error" is the known Bun 1.2.18 teardown phantom + flaky Biome test, unrelated — see #400).
- `bun run typecheck` clean; `bunx biome check .` exit 0.
- **Dogfood: 29/29 addons still pass `lint-capability` unchanged.**

## Review notes

This PR also fixes two real defects found during orchestrator self-review of the initial implementation: (a) the `return`-keyword regex detection was dead because the trailing-whitespace before `/` defeated the identifier re-read; (b) a block comment overwrote the heuristic's `prevSignificant`, contradicting its own "non-comment" contract. Both are covered by new regression tests.

**Cross-model review:** local `gemini` CLI is quota-exhausted (resets ~2h) and `codex` requires node (blocked by repo hook); per the established fallback, the async PR review bots (gemini-code-assist + CodeRabbit) serve as the cross-model pass — their findings will be addressed before merge.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive test coverage for comment handling and import extraction in the linting system, including edge cases with strings and template literals.

* **Refactor**
  * Enhanced comment detection algorithm to provide more robust and reliable code analysis, improving overall linting accuracy.

[Review Change Stack](https://app.coderabbit.ai/change-stack/laofahai/linchkit/pull/417?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: packages/devtools/src/methodology/capability-lint.ts [hunk 1]
```diff
@@ -315,28 +315,202 @@ function isIgnoredDir(name: string): boolean {
...
+    "in",
+    "of",
+    "do",
+    "else",
+    "yield",
+    "void",
+    "delete",
+    "instanceof",
+    "new",
+  ]);
```
[the remark this answers is not in the document]
Comment (laofahai) [reply packages/devtools/src/methodology/capability-lint.ts:376]: Done in 7a90256 — added `throw` and `default` to `regexKeywords`. `throw /e/` and `export default /re/` are now recognized as regex contexts, so a trailing comment on those lines is correctly stripped. Added a regression test for the `throw` path.

File: packages/devtools/src/methodology/capability-lint.ts [hunk 2]
```diff
@@ -315,28 +315,202 @@ function isIgnoredDir(name: string): boolean {
...
+
+  const isExprEndChar = (c: string): boolean =>
+    // Identifier/number chars, a closing bracket, or a string/regex end can
+    // terminate an expression → a following `/` is division, not a regex.
+    /[\w$)\]}'"`]/.test(c);
```
[the remark this answers is not in the document]
Comment (laofahai) [reply packages/devtools/src/methodology/capability-lint.ts]: Done in 7a90256 — rewrote `isExprEndChar` to test the regex-permitting punctuator set by EXCLUSION (`!/[=,({[;:!&|?+\-*/%<>^~]/`). Now any non-punctuator char — including Unicode identifiers, not just ASCII `\w` — reads as expression-ending, so a `/` after a Unicode-named variable stays division and its trailing comment is stripped. Added a regression test using a Chinese-named identifier.
